Simplenote/Classes/SPNoteEditorViewController.m

```diff
@@ -217,9 +210,42 @@ - (void)viewWillAppear:(BOOL)animated {
 - (void)viewDidAppear:(BOOL)animated
 {
     [super viewDidAppear:animated];
+
+    if (!_currentNote) {
+        [self newButtonAction:nil];
+    } else {
+        [_noteEditorTextView processChecklists];
+        self.userActivity = [NSUserActivity openNoteActivityFor:_currentNote];
+    }
```

Comment:
viewWillAppear is called very early during the state restoration process, earlier than decodeRestorableStateWithCoder is called.  It is in decodeRestorableStateWithCoder that we restore the _currentNote.  So if this code needs to be done later or else the _currentNote will be nil and a new note is created instead of the old note restored.

This also results in the UITableView row count mismatch warning that I'm sure you've seen.

Reply:
(not a great fan of the way existing/new notes are determined but I'm trying to keep a light touch on this patch)

Reply:
Thanks for the explanation sir!! It's been years since I did state restoration 😃